### PR TITLE
Update robots.txt for Improved SEO and Documentation Visibility

### PR DESCRIPTION
**Pull request by loadinglit**
This pull request addresses fixes #1960 of taipy by updating the robots.txt file in the taipy-doc repository to enhance search engine optimization (SEO) and user experience.

- SEO Improvement: Allowed search engines to index the latest documentation and version 4.0, ensuring users can easily access
the most up-to-date resources.
- Disallowed indexing of older documentation versions (3.0, 3.1, 2.0, 2.1, 2.2, 2.3, 2.4) to reduce confusion and direct traffic to the current content.

These modifications are essential for maintaining clarity in documentation access and ensuring users find the most relevant information.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -81,6 +81,7 @@ target/
 .venv
 ENV/
 venv/
+os/
```
Comment (FabienLelaquais) [inline .gitignore]: What do we need that for?

File: docs/robots.txt [hunk 2]
```diff
@@ -1,5 +1,12 @@
 User-agent: *
 Allow: /en/latest/ # Allow all robots to access the latest version of the documentation
-Disallow: /en/ # Disallow all robots to access the other versions of the documentation
+Allow: /en/release-4.0/ # Allow robots to access version 4.0, which is a newer version
```
Comment (FabienLelaquais) [inline docs/robots.txt]: Excellent!

**Review by FabienLelaquais — APPROVED** [review]
Great!

File: .gitignore (hunk 1 above)
Comment (loadinglit) [reply .gitignore]: Hey, pls check.. I removed the unnecessary change of this

File: docs/robots.txt (hunk 2 above)
Comment (jrobinAV) [reply docs/robots.txt]: Should we disallow release-4.0 as well since we want google to only parse latest version?

Comment (loadinglit) [reply docs/robots.txt]: > Should we disallow release-4.0 as well since we want google to only parse latest version?

yes agreed, will make changes

Comment (FabienLelaquais) [reply docs/robots.txt]: Agreed!!

File: docs/robots.txt [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+Disallow: /en/release-3.0/ # Disallow older version 3.0 to avoid confusion
```
Comment (FabienLelaquais) [inline docs/robots.txt]:
```suggestion
Disallow: /en/release-3.0/ # Disallow all legacy versions to enforce latest
```

File: .gitignore
Comment (FabienLelaquais) [inline .gitignore:1]: What? You deleted this file?!? Why???

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
.gitignore is gone?

File: .gitignore
Comment (loadinglit) [reply .gitignore:1]: sorry that was a mistake  it is added now

**Comment by loadinglit** [comment]
@jrobinAV @FabienLelaquais 
- Disallowed release version 4.0 in robots.txt
- fixed the faulty irregularity of gitignore

**Comment by jrobinAV** [comment]
Another Pull request has been proposed before this one to solve the same issue. 
https://github.com/Avaiga/taipy-doc/pull/1154

I propose to close this one.

**Pull request closed without merging**